packages/modules/common/fault_state.py

```diff
@@ -0,0 +1,57 @@
+from enum import Enum
+import traceback
+
+try:
+    from ...helpermodules import compatibility
+    from ...helpermodules import log
+    from ...helpermodules import pub
+except (ImportError, ValueError, SystemError):
+    # for 1.9 compatibility
+    from helpermodules import compatibility
+    from helpermodules import log
+    from helpermodules import pub
+
+
+class FaultStateLevel(Enum):
+    NO_ERROR = 0
+    WARNING = 1
+    ERROR = 2
+
+
+class ComponentInfo:
+    def __init__(self, id: int, name: str, type: str) -> None:
+        self.id = id
+        self.name = name
+        self.type = type
+
+
+class FaultState(Exception):
```

Comment:
Der Name `FaultState` passt tatsächlich deutlich besser. Aber dann ist es merkwürdig, dass die Klasse von `Exception` erbt und auch als Exception verwendet wird.

Die Klasse macht auch zwei Dinge gleichzeitig: Sie dokumentiert einen Fehlerstatus und übernimmt die Speicherung des Status. Damit verletzt sie das "principle of single responsibility".

Mein Vorschlag wäre die Funktion `store_error` aus dieser Klasse rauszutrennen. Also entweder einfach nur in eine globale Funktion, oder in eine neue Klasse `ErrorStore` oder sowas in der Art. Dann kann man auch global `store_error(None, component)` (oder sowas in der Richtung) aufrufen ohne dass man dafür die Klasse `FaultState` dafür braucht. Die Konstante `FaultStateLevel.NO_ERROR` würde dann auch nicht mehr gebraucht. Dann könnte auch der alte Name `ModuleError` (vielleicht besser noch `ComponentError`) bleiben.

Wie es ansonsten mit der Klasse weiter geht hängt erheblich an [meinem Kommentar zu Exceptiontypen mit Struktur](https://github.com/snaptec/openWB/pull/1690#discussion_r755367571). Eine richtige Exception-Hierarchie bietet letztlich deutlich mehr Möglichkeiten.